@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1772,16 +1782,9 @@ export class TextElementsRenderer {
                         iconRenderState.startFadeIn(mapViewState.frameNumber, mapViewState.time);
                     }
                 }
-            }
-            // If the icon is prepared and valid, but just not visible, try again next time.
```

Comment:
Moved this branch of the if to the top to do early exit, trying to reduce the nesting levels here.